### ISSUE TITLE
ConfigError: Config "antfu/javascript/rules": Key "rules": Key "accessor-pairs": structuredClone is not defined

### ISSUE THREAD

**Issue by liu-xinhui** [antfu/eslint-config#579]
### Describe the bug

A brand new Vite Vue project is reporting an error without any modifications.

```
Oops! Something went wrong! :(

ESLint: 9.5.0

ConfigError: Config "antfu/javascript/rules": Key "rules": Key "accessor-pairs": structuredClone is not defined
    at rethrowConfigError (/Users/liuxinhui/java/模板20240806/vite-project/node_modules/.pnpm/@eslint+config-array@0.16.0/node_modules/@eslint/config-array/dist/cjs/indecjs:302:8)
    at /Users/liuxinhui/java/模板20240806/vite-project/node_modules/.pnpm/@eslint+config-array@0.16.0/node_modules/@eslint/config-array/dist/cjs/index.cjs:1097:5
    at Array.reduce (<anonymous>)
    at FlatConfigArray.getConfigWithStatus (/Users/liuxinhui/java/模板20240806/vite-project/node_modules/.pnpm/@eslint+config-array@0.16.0/node_modules/@eslint/config-ary/dist/cjs/index.cjs:1090:43)
    at FlatConfigArray.getConfig (/Users/liuxinhui/java/模板20240806/vite-project/node_modules/.pnpm/@eslint+config-array@0.16.0/node_modules/@eslint/config-array/dist/c/index.cjs:1119:15)
    at /Users/liuxinhui/java/模板20240806/vite-project/node_modules/.pnpm/eslint@9.5.0/node_modules/eslint/lib/eslint/eslint-helpers.js:346:56
    at Array.reduce (<anonymous>)
    at /Users/liuxinhui/java/模板20240806/vite-project/node_modules/.pnpm/eslint@9.5.0/node_modules/eslint/lib/eslint/eslint-helpers.js:333:36
    at /Users/liuxinhui/java/模板20240806/vite-project/node_modules/.pnpm/eslint@9.5.0/node_modules/eslint/lib/eslint/eslint-helpers.js:296:32
    at Object.isAppliedFilter (/Users/liuxinhui/java/模板20240806/vite-project/node_modules/.pnpm/@nodelib+fs.walk@1.2.8/node_modules/@nodelib/fs.walk/out/readers/commons:12:31)
```
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/de6f7d61-81d3-48ba-8246-e084c0e1cbc3">
<img width="1159" alt="image" src="https://github.com/user-attachments/assets/524db915-1087-48db-a229-9b25b488d3be">

### Reproduction

`pnpm create vite` and then `pnpm dlx @antfu/eslint-config@latest`

### System Info

```Shell
System:
    OS: macOS 13.6
    CPU: (8) arm64 Apple M1 Pro
    Memory: 89.20 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.20.2 - /usr/local/bin/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn
    npm: 8.19.4 - /usr/local/bin/npm
    pnpm: 8.12.1 - /usr/local/bin/pnpm
  Browsers:
    Chrome: 127.0.6533.100
    Safari: 16.6
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### Contributions

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests (actually just go ahead and do it, thanks!)

**Comment by liu-xinhui** [comment]
this issues fixed after I upgrade node to 20.16.0
